File: Nexxy-Frontend/src/app/Entities/movie/Movie.ts
export interface Movie {
  adult: boolean;
  backdrop_path: string;
  genre_ids: number[];
  id: number;
  name?:string;
  media_type: string;
  original_language: string;
  original_title: string;
  original_name?:string;
  overview: string;
  popularity: number;
  poster_path: string;
  release_date: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
  showVideo?: boolean;
}

export class MovieImpl implements Movie {
  adult: boolean;
  backdrop_path: string;
  genre_ids: number[];
  id: number;
  media_type: string;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string;
  release_date: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
  showVideo?: boolean;
  name: string |undefined;
  original_name?: string | undefined;

  constructor() {
    (this.adult = false),
      (this.backdrop_path = ''),
      (this.genre_ids = []),
      (this.id = 0),
      (this.media_type = ''),
      (this.original_language = ''),
      (this.original_title = ''),
      (this.overview = ''),
      (this.popularity = 0),
      (this.poster_path = ''),
      (this.release_date = ''),
      (this.title = ''),
      (this.video = false),
      (this.vote_average = 0),
      (this.vote_count = 0),
      (this.showVideo = false);
  }
}

File: Nexxy-Backend/src/main/java/com/nexxy/entities/movie/Movie.java
package com.nexxy.entities.movie;

import jakarta.persistence.*;
import org.hibernate.annotations.Array;

import java.util.Date;
import java.util.List;

@Entity
public class Movie {
    @Id
    private Long id;

    private List<Long> genre_ids;

    private Boolean adult;

    private String backdrop_path;

    private String media_type;
    private String original_language;
    private String original_title;
    private String overview;
    private String poster_path;
    private String title;
    private String release_date;
    private Boolean video;
    private Double popularity;
    private Double vote_average;
    private Long vote_count;


}
